WorkloadManagementSystem/JobWrapper/Watchdog.py

```diff
@@ -20,22 +20,45 @@
 __RCSID__ = "$Id$"
 
 import os
+import re
 import time
 
 from DIRAC                                              import S_OK, S_ERROR, gLogger
 from DIRAC.Core.Utilities                               import Time
+from DIRAC.Core.Utilities                               import MJF
 from DIRAC.Core.DISET.RPCClient                         import RPCClient
 from DIRAC.ConfigurationSystem.Client.Config            import gConfig
 from DIRAC.ConfigurationSystem.Client.PathFinder        import getSystemInstance
 from DIRAC.Core.Utilities.ProcessMonitor                import ProcessMonitor
 from DIRAC.Core.Utilities.TimeLeft.TimeLeft             import TimeLeft
+from DIRAC.Core.Utilities.Subprocess                    import getChildrenPIDs
 
 class Watchdog( object ):
 
   #############################################################################
-  def __init__( self, pid, exeThread, spObject, jobCPUtime, memoryLimit = 0, processors = 1, systemFlag = 'linux' ):
+  def __init__( self, pid, exeThread, spObject, jobCPUtime, jobArgs, memoryLimit = 0, processors = 1, systemFlag = 'linux' ):
```

Comment:
This is the only thing that makes me a bit scared. I would prefer that the change of signature would be:

```
def __init__( self, pid, exeThread, spObject, jobCPUtime, memoryLimit = 0, processors = 1, systemFlag = 'linux', jobArgs = {} ):
```